src/collection.ts

```diff
@@ -1652,16 +1649,30 @@ export class Collection implements OperationParent {
    * @param options - Optional settings for the command
    * @param callback - An optional callback, a Promise will be returned if none is provided
    */
+  findAndRemove(query: Document, callback: Callback): void;
+  findAndRemove(query: Document, sort: Sort, callback: Callback): void;
+  findAndRemove(query: Document, options: FindAndModifyOptions, callback: Callback): void;
   findAndRemove(
     query: Document,
-    sort: Document,
+    sort: Sort,
     options: FindAndModifyOptions,
     callback: Callback
-  ): Promise<Document> | void {
-    const args = Array.prototype.slice.call(arguments, 1);
-    callback = typeof args[args.length - 1] === 'function' ? args.pop() : undefined;
-    sort = args.length ? args.shift() || [] : [];
-    options = args.length ? args.shift() || {} : {};
+  ): void;
+  findAndRemove(query: Document): Promise<Document>;
+  findAndRemove(query: Document, sort: Sort): Promise<Document>;
+  findAndRemove(query: Document, sort: Sort, options: FindAndModifyOptions): Promise<Document>;
+  findAndRemove(
+    query: Document,
+    ...args: any[]
+  ): // TODO: Use labeled tuples when api-extractor supports TS 4.0
+  /*sort?: Sort | FindAndModifyOptions | Callback<Document>, */
+  /*options?: FindAndModifyOptions | Callback<Document>, */
+  /*callback?: Callback<Document>*/
```

Comment:
At first glance this might look like comment garbage, and I can get rid of it if we don't want this to potentially sit here and get stale.
I would like to type args with
```typescript
...args: [sort?: Sort | FindAndModifyOptions | Callback<Document>, options?: FindAndModifyOptions | Callback<Document>, callback?: Callback<Document>]
```
Making use of the new labeled tuples gives a friendly auto complete and allows "optional" tuple members with the `?:` symbol, unfortunately you can't replicate that with a plain tuple, and currently api-extractor doesn't have the logic to parse this new TS feature.

Reply:
You do still make use of arguments in distinct to check the number of arguments provided. Doing a spread just makes this array explicit rather than the implicit global. I do agree though there isn't a benefit larger than that for now while we can't do the labeled tuples but they do provide nice hinting so something to consider for the future.